### PR TITLE
Recursive autotranslation

### PR DESCRIPTION
**Pull request by sven15**
automatically translates not yet translated parent posts

### CODE REVIEW

File: wp-content/plugins/automatic-translation/TranslationManager.php [hunk 1]
```diff
@@ -114,6 +118,20 @@ private function create_translation($post, $source_language_code, $target_langua
 		if ($current_translation_id !== null) {
 			$translated_post['ID'] = $current_translation_id;
 		}
+		
+		// check if translation of possible parent posts exists before creating translation
+		$parent_id = $this->translated_parent_exists( $post, $target_language_code );
+		if( ! $parent_id ) {
+
```
Comment (Meash) [inline wp-content/plugins/automatic-translation/TranslationManager.php]: why the new lines? keep it easily readable

File: wp-content/plugins/automatic-translation/TranslationManager.php [hunk 2]
```diff
@@ -121,6 +139,15 @@ private function create_translation($post, $source_language_code, $target_langua
 		$this->add_save_post_hook();
 		$this->wpml_helper->link_wpml($post->ID, $post->post_type, $translated_post_id, $target_language_code);
 		$this->mark_as_automatic_translation($translated_post_id);
+		
+		//fixes wrong post_parent for autotranslation
```
Comment (Meash) [inline wp-content/plugins/automatic-translation/TranslationManager.php]: can you elaborate on what exactly is being fixed here?

File: wp-content/plugins/automatic-translation/TranslationWpmlHelper.php [hunk 3]
```diff
@@ -19,4 +19,48 @@ public function link_wpml($source_post_id, $source_post_type, $translated_post_i
 		}
 		$sitepress->set_element_language_details($translated_post_id, $wpml_post_type, $source_trid, $language_code);
 	}
-}
\ No newline at end of file
+	
+	//not needed yet
```
Comment (Meash) [inline wp-content/plugins/automatic-translation/TranslationWpmlHelper.php]: then why does it exist?

File: wp-content/plugins/automatic-translation/TranslationWpmlHelper.php [hunk 4]
```diff
@@ -19,4 +19,48 @@ public function link_wpml($source_post_id, $source_post_type, $translated_post_i
...
+	public function get_translation_post_parent( $post_id, $source_language_code, $target_language_code ) {
+		global $wpdb;
+
```
Comment (Meash) [inline wp-content/plugins/automatic-translation/TranslationWpmlHelper.php]: too many newlines

File: wp-content/plugins/automatic-translation/TranslationWpmlHelper.php [hunk 5]
```diff
@@ -19,4 +19,48 @@ public function link_wpml($source_post_id, $source_post_type, $translated_post_i
...
 	}
-}
\ No newline at end of file
+	
+	//not needed yet
+	public function get_translation_post_parent( $post_id, $source_language_code, $target_language_code ) {
+		global $wpdb;
+		
+		
+		//convert single querys to join!
```
Comment (Meash) [inline wp-content/plugins/automatic-translation/TranslationWpmlHelper.php]: there is no reason to join manually here. let SQL do that for you, i.e. 

    SELECT post_parent, post_status 
    FROM $wpdb->posts posts
    JOIN {$wpdb->prefix}icl_translations translations_id ON translations_id.element_id = '$post_id'
    JOIN {$wpdb->prefix}icl_translations translations_source  ON translations_source.element_id = posts.ID AND language_code = '$source_language_code' AND translations_source.trid = translations_id.trid

(untested)

File: wp-content/plugins/automatic-translation/TranslationManager.php (hunk 1 above)
Comment (sven15) [inline wp-content/plugins/automatic-translation/TranslationManager.php]: Keeping it easily readable by structuring logical parts of the code? :-)

File: wp-content/plugins/automatic-translation/TranslationManager.php (hunk 2 above)
Comment (sven15) [inline wp-content/plugins/automatic-translation/TranslationManager.php]: I found earlier, that you can fix the problem where a page is not belonging to its parent by clicking on publish again. And doing the update also works. I cannot say if this is a bug in WPML, Wordpress or our plugin.

File: wp-content/plugins/automatic-translation/TranslationWpmlHelper.php (hunk 3 above)
Comment (sven15) [inline wp-content/plugins/automatic-translation/TranslationWpmlHelper.php]: forgot to remove it :-)

File: wp-content/plugins/automatic-translation/TranslationWpmlHelper.php (hunk 5 above)
Comment (sven15) [inline wp-content/plugins/automatic-translation/TranslationWpmlHelper.php]: Yeah, something like that.

File: wp-content/plugins/automatic-translation/TranslationManager.php (hunk 2 above)
Comment (sven15) [inline wp-content/plugins/automatic-translation/TranslationManager.php]: With calling $this->wpml_helper->get_translation_post_parent one wp_update_post is enough.

Comment (Meash) [inline wp-content/plugins/automatic-translation/TranslationManager.php]: Please put that explanation into the code comment as well :)

**Pull request merged**